### PR TITLE
7.0 monitoring

### PR DESCRIPTION
**Pull request by gurneyalex**
port MP from https://code.launchpad.net/~camptocamp/server-env-tools/7.0-monitoring/+merge/215138 as requested as part of #35

### CODE REVIEW

**Comment by gurneyalex** [comment]
@pedrobaeza you had comments as part of the Launchpad MP. Some of them are done. 

as for the others: 
- "L685: the 'info' field is either "rpc call" or "cron job". Wouldn't a selection field be more efficient here?" -> I don't think so, as selection fields are not implemented with a foreign table, so you end up with a string in db + a check on the validity of the value
- "L756: option to disable Server Process log_measure is not available yet, correct?" -> correct
- "L771: I worry about this extra search to translate model_name into model_id. Why not store model_name directly?" -> storing an integer instead of a string is a lot more efficient, and it allows using the translations later on, and making joins. 
- "L778: I see that 'status' is kept empty; is this intended to store WARNING and ERROR results? That would be an excellent addition." -> it was originally thought for storing the HTTP status of the xmlrpc query. I'm not sure how to get this, so I'm leaving empty for now. 
- "L1146: for Process logs, would it be possible to add an option to group by day? (currently group by month, which is insufficient...)" -> I'm not sure how to do this. Contributions / help welcome.

**Comment by gurneyalex** [comment]
travis is red because of flake8 issues in other addons. #38 fixes these.

**Comment by pedrobaeza** [comment]
Thanks for the changes, @gurneyalex. For now, is enough. This is a great improvement from having nothing :wink: Afterwards, I will make another PR trying to improve current module.

:+1: 

Regards.

**Comment by hbrunn** [comment]
:+1:

File: server_monitoring/server_monitoring.py [hunk 1]
```diff
@@ -0,0 +1,447 @@
+# -*- coding: utf-8 -*-
```
Comment (bwrsandman) [inline server_monitoring/server_monitoring.py]: This script is far too long, could you please split it per class/model.

File: server_monitoring/server_monitoring_view.xml [hunk 2]
```diff
@@ -0,0 +1,342 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (bwrsandman) [inline server_monitoring/server_monitoring_view.xml]: Same as with the python script, can you split this file by model, the larger parts, at least.

File: server_monitoring/server_monitoring.py [hunk 3]
```diff
@@ -0,0 +1,447 @@
...
+TODO: check if we can log pooler state (whatever that is...)
+"""
+
+import logging
+import gc
+from operator import itemgetter
+import types
+import os
+import threading
+import resource
```
Comment (legalsylvain) [inline server_monitoring/server_monitoring.py]: Hi, 
I have some bad log, installing your module.

```
Ambiguous import: the OpenERP module `resource` is shadowed by another
module (available at /usr/lib/python2.7/lib-dynload/resource.so).
To import it, use `import openerp.addons.<module>.`.
```

File: server_monitoring/__openerp__.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        'server_monitoring_view.xml',
+        'server_monitoring_data.xml',
+        'security/ir.model.access.csv',
+        ],
+    'test': [
+        'tests/test_monitor_process.yml',
+        'tests/test_monitor_database.yml',
+        ],
+    'installable': True,
+    'application': False,
```
Comment (legalsylvain) [inline server_monitoring/__openerp__.py]: (Minor comment)
L57: 'True' is the default value, is it necessary ?
L58: 'False' is the default value, is it necessary ?

**Comment by legalsylvain** [comment]
Hi, 
Just finish to test this module and it's very usefull. Thanks a lot !
:+1: (except the import problem)
(Functional review)

**Comment by hbrunn** [comment]
I'll label this 'needs fixing' for Sandy's points

**Comment by dreispt** [comment]
@gurneyalex did you figure out anything about those SERIALIZATION_FAILURE errors I found before?

File: server_monitoring/server_monitoring.py (hunk 3 above)
Comment (gurneyalex) [inline server_monitoring/server_monitoring.py]: There's nothing much I can do about this. What I need here is the stdlib resource module, not openerp.addons.resource. This warning is caused by Odoo playing tricks with sys.path to allow shortcut imports, as well as importing from 'odoo' or 'openerp' indifferently.

**Comment by gurneyalex** [comment]
@dreispt I'm a bit under the water atm and did not get a chance to work on this addon. I'll give this a careful look Real Soon[tm].

From what I remember, these errors where logged but the actions would be performed anyway and retried. But I need to double check.

File: server_monitoring/__openerp__.py (hunk 4 above)
Comment (dreispt) [inline server_monitoring/__openerp__.py]: L57 is better left as is, since it allows a find/replace to disable modules.
L58 is unnecessary.

**Comment by pedrobaeza** [comment]
@gurneyalex, can you check last comments and I'll merge?

**Comment by gurneyalex** [comment]
@dreispt @pedrobaeza I've updated the addon, split the large files in smaller ones, added a README.rst, squashed the commits and fixed the CI. 

Regarding the serialization failures, I think this should be solved by installing `base_concurrency` (in OCA/server-tools). If you feel it's worth it, I can add a dependency.

File: server_monitoring/README.rst [hunk 5]
```diff
@@ -0,0 +1,88 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :alt: License: AGPL-3
+
+server monitoring
+==============
```
Comment (pedrobaeza) [inline server_monitoring/README.rst]: This should be of the same length

Comment (pedrobaeza) [inline server_monitoring/README.rst]: You can add the same line on top to make the title more important (see https://github.com/OCA/maintainer-tools/pull/109)

File: server_monitoring/README.rst [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+=====
+
+Just install the addon in your instance. You will be able to access the logs
+under Reporting -> Server Monitoring.
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/{repo_id}/{branch}
+
+.. repo_id is available in https://github.com/OCA/maintainer-tools/blob/master/tools/repos_with_ids.txt
```
Comment (pedrobaeza) [inline server_monitoring/README.rst]: Remove these lines

File: server_monitoring/README.rst [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+
+Usage
+=====
+
+Just install the addon in your instance. You will be able to access the logs
+under Reporting -> Server Monitoring.
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/{repo_id}/{branch}
```
Comment (pedrobaeza) [inline server_monitoring/README.rst]: Replace {repo_id}/{branch} with the proper information

File: server_monitoring/README.rst [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+* enhance group by for process logs. 
+
+
+Bug Tracker
+===========
+
+Bugs are tracked on `GitHub Issues <https://github.com/OCA/{project_repo}/issues>`_.
+In case of trouble, please check there if your issue has already been reported.
+If you spotted it first, help us smashing it by providing a detailed and welcomed feedback
+`here <https://github.com/OCA/{project_repo}/issues/new?body=module:%20{module_name}%0Aversion:%20{version}%0A%0A**Steps%20to%20reproduce**%0A-%20...%0A%0A**Current%20behavior**%0A%0A**Expected%20behavior**>`_.
```
Comment (pedrobaeza) [inline server_monitoring/README.rst]: replace placeholders for their values

File: server_monitoring/README.rst [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+* check if we can / wish to log session id
+* check if we can log pooler state (whatever that is...)
+* check if we can log HTTP status
+* enhance group by for process logs. 
+
+
+Bug Tracker
+===========
+
+Bugs are tracked on `GitHub Issues <https://github.com/OCA/{project_repo}/issues>`_.
```
Comment (pedrobaeza) [inline server_monitoring/README.rst]: Replace placeholder by server-tools

**Comment by gurneyalex** [comment]
expecting 3 :green_heart: CI checks anytime now

**Comment by gurneyalex** [comment]
I'll port this addon to 8.0

File: server_monitoring/README.rst [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+=====
+
+Just install the addon in your instance. You will be able to access the logs
+under Reporting -> Server Monitoring.
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/149/7.0
+
+.. repo_id is available in https://github.com/OCA/maintainer-tools/blob/master/tools/repos_with_ids.txt
```
Comment (pedrobaeza) [inline server_monitoring/README.rst]: Remove these lines, as they are guidelines for the template

File: server_monitoring/__openerp__.py [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+{
+    'name': 'Server Monitoring',
+    'version': '0.1',
+    'category': 'Connector',
```
Comment (pedrobaeza) [inline server_monitoring/__openerp__.py]: Why connector category?

**Comment by pedrobaeza** [comment]
Little remark, but :+1: functional and technical

**Comment by max3903** [comment]
:+1:

**Comment by moylop260** [comment]
:+1:

**Pull request merged**